Source/WebCore/dom/Element.cpp

```diff
@@ -2141,6 +2123,25 @@ void Element::attributeChanged(const QualifiedName& name, const AtomString& oldV
         cache->deferAttributeChangeIfNeeded(this, name, oldValue, newValue);
 }
 
+void Element::setEffectiveLangInSubtree()
```

Comment:
How about recursivelyUpdateEffectiveLang?
This function doesn't necessarily update the effective lang for the whole subtree, skipping any sub-subtree with a lang attribute of its own so it's more fitting to me to put the emphasis on recursive nature of the algorithm instead of subtree.

Or maybe updateEffectiveLangOnDescendants.